src/function/scalar/generic/current_setting.cpp

```diff
@@ -39,13 +39,19 @@ unique_ptr<FunctionData> CurrentSettingBind(ClientContext &context, ScalarFuncti
 		throw ParserException("Key name for struct_extract needs to be neither NULL nor empty");
 	}
 
-	auto &config_map = context.db->config.set_variables;
-	if (config_map.find(key_val.str_value) == config_map.end()) {
-		throw InvalidInputException("Variable '%s' was not SET in this context", key_val.str_value);
+	const auto &key = key_val.str_value;
+	auto &session_config_map = context.set_variables;
+	auto &global_config_map = context.db->config.set_variables;
+
+	bool has_session_value = session_config_map.find(key) != session_config_map.end();
+	bool has_global_value = global_config_map.find(key) != global_config_map.end();
+	if (has_session_value || has_global_value) {
```

Comment:
Could we extract this behavior into a function (e.g. `bool ClientContext::TryGetSetting(const string &name, Value &result)`), so we can use it for e.g. the HTTPFS reader?

Reply:
Sure! Done.